ebpf/tracer_test.go

```diff
@@ -559,6 +602,46 @@ func BenchmarkTCPSend(b *testing.B) {
 	runBenchtests(b, payloadSizesTCP, "eBPF", benchSendTCP)
 }
 
+func benchTCPClose(count int, t *Tracer) func(b *testing.B) {
+	collect := func() (*Connections, error) { return nil, nil }
+
+	if t != nil {
+		collect = t.GetActiveConnections
+	}
+
+	echoOnMessage := func(c net.Conn) {
+		r := bufio.NewReader(c)
+		for {
+			buf, err := r.ReadBytes(byte('\n'))
+			if err == io.EOF {
+				c.Close()
+				return
+			}
+			c.Write(buf)
+		}
+	}
+
+	return func(b *testing.B) {
+		end := make(chan struct{})
+		server := NewTCPServer(echoOnMessage)
+		server.Run(end)
+		for i := 0; i < count; i++ {
```

Comment:
This is not how you're supposed to use the benchmark library. 

You can't choose how many iterations for it to run. 

You need to use `b.N`, and the benchmarking library will run your test as many times as it can in 1 second, and divide the number of iterations over that period to figure out time per iteration.

You can however, change the how long the benchmark will run for (via `-benchtime 30s`)

Reply:
Thanks ! I just updated the benchmarks 👍